index.js

```diff
@@ -25,6 +25,9 @@ module.exports = (paths, options) => pTry(() => {
 	}
 
 	paths = paths
+		.filter(filePath => !paths.some(
+			otherPath => filePath !== otherPath && path.dirname(filePath) === otherPath)
+		)
```

Comment:
This needs to be after the `.map` as it will only work correctly after the path is made absolute.